source/adapters/cuda/command_buffer.hpp

```diff
@@ -172,12 +172,18 @@ struct usm_memcpy_command_handle : ur_exp_command_buffer_command_handle_t_ {
 struct usm_fill_command_handle : ur_exp_command_buffer_command_handle_t_ {
   usm_fill_command_handle(ur_exp_command_buffer_handle_t CommandBuffer,
                           CUgraphNode Node, CUgraphNode SignalNode,
-                          std::vector<CUgraphNode> WaitNodes)
+                          std::vector<CUgraphNode> WaitNodes,
+                          const std::vector<CUgraphNode> &DecomposedNodes = {})
       : ur_exp_command_buffer_command_handle_t_(CommandBuffer, Node, SignalNode,
-                                                WaitNodes) {}
+                                                WaitNodes),
+        DecomposedNodes(std::move(DecomposedNodes)) {}
   CommandType getCommandType() const noexcept override {
     return CommandType::USMFill;
   }
+
+  // If this fill command was decomposed into multiple nodes, this vector
+  // contains all of those nodes in the order they were added to the graph.
+  std::vector<CUgraphNode> DecomposedNodes;
```

Comment:
Do we need to store this, do we ever reference this member?

Reply:
Yes I figured since they are created as a separate vector now and we will need them in future to update a decomposed fill command that it made sense to just store them. Since the CUDA adapter already has some future proofing for updating different types of commands it seemed easy to just store them here for now.

I've updated the comments for these members to reflect this.